create_update_image.sh

```diff
@@ -1,14 +1,187 @@
+	local _seeking="$1"
+	local _idx=0
+	for element in "${_allowed[@]}"
+	do
+		test "$element" = "$_seeking" && { test "$3" = "idx" && echo "$_idx" || echo "$element"; } && return 0
+		_idx=$((_idx + 1))
+	done
+	die "Value '$_seeking' (of argument '$2') doesn't match the list of allowed values: 'download_rpms', 'install_rpms', 'install_addon_from_repo', 'create_image' and 'cleanup'" 4
+}
+
+# THE DEFAULTS INITIALIZATION - POSITIONALS
+_positionals=()
+_arg_start_with="install_rpms"
+# THE DEFAULTS INITIALIZATION - OPTIONALS
+_arg_tmp_root="$(mktemp -d)"
+_arg_rpm_dir="$build_dir/rpm"
+
+print_help ()
+{
+	printf 'Usage: %s [--tmp-root <arg>] [--rpm-dir <arg>] [-h|--help] [<start-with>]\n' "$0"
+	printf '\t%s\n' "<start-with>: Action to start with (default: 'install_rpms')"
+	printf '\t%s\n' "--tmp-root: Fake temp root (default: '$(mktemp -d)')"
+	printf '\t%s\n' "--rpm-dir: Where to put/take from RPMs to install (default: '$build_dir/rpm')"
+	printf '\t%s\n' "-h,--help: Prints help"
+}
+
+parse_commandline ()
+{
+	while test $# -gt 0
+	do
+		_key="$1"
+		case "$_key" in
+			--tmp-root)
+				test $# -lt 2 && die "Missing value for the optional argument '$_key'." 1
+				_arg_tmp_root="$2"
+				shift
+				;;
+			--tmp-root=*)
+				_arg_tmp_root="${_key##--tmp-root=}"
+				;;
+			--rpm-dir)
+				test $# -lt 2 && die "Missing value for the optional argument '$_key'." 1
+				_arg_rpm_dir="$2"
+				shift
+				;;
+			--rpm-dir=*)
+				_arg_rpm_dir="${_key##--rpm-dir=}"
+				;;
+			-h|--help)
+				print_help
+				exit 0
+				;;
+			-h*)
+				print_help
+				exit 0
+				;;
+			*)
+				_positionals+=("$1")
+				;;
+		esac
+		shift
+	done
+}
+
+
+handle_passed_args_count ()
+{
+	test ${#_positionals[@]} -le 1 || _PRINT_HELP=yes die "FATAL ERROR: There were spurious positional arguments --- we expect between 0 and 1, but got ${#_positionals[@]} (the last one was: '${_positionals[*]: -1}')." 1
+}
+
+assign_positional_args ()
+{
+	_positional_names=('_arg_start_with' )
+
+	for (( ii = 0; ii < ${#_positionals[@]}; ii++))
+	do
+		eval "${_positional_names[ii]}=\${_positionals[ii]}" || die "Error during argument parsing, possibly an Argbash bug." 1
+	done
+}
+
+parse_commandline "$@"
+handle_passed_args_count
+assign_positional_args
+
+# OTHER STUFF GENERATED BY Argbash
+# Validation of values
+_arg_start_with="$(action "$_arg_start_with" "start-with")" || exit 1
+_arg_start_with_index="$(action "$_arg_start_with" "start-with" idx)"
+
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+# [ <-- needed because of Argbash
+
+tmp_root="$_arg_tmp_root"
+rpmdir="$_arg_rpm_dir"
+
+
+packages="
+	openscap
+	openscap-python3
+	openscap-scanner
+	python3-cpio
+	python3-pycurl
+	oscap-anaconda-addon
+"
+
+
+actions=(download_rpms install_rpms install_addon_from_repo create_image cleanup)
+
+
+download_rpms() {
+	mkdir -p "$rpmdir"
+	(cd "$rpmdir" && dnf download --arch x86_64,noarch $packages)
```

Comment:
In the future the download may need to be improved to fetch packages from other releases.
I am using Fedora 27 so packages downloaded were fc27, but image installed was a Fedora 28.
It didn't matter and installation worked, I wonder if it can be a problem one day.

Reply:
I have added a releasever option, defaulting to 28. Good catch!